src/sdam/description/topology/mod.rs

```diff
@@ -204,27 +205,20 @@ impl TopologyDescription {
         command: &mut Command<T>,
         criteria: Option<&SelectionCriteria>,
     ) {
-        match criteria {
-            Some(SelectionCriteria::ReadPreference(ReadPreference::Secondary { ref options })) => {
-                command.set_read_preference(ReadPreference::Secondary {
-                    options: options.clone(),
-                })
+        let read_preference = match criteria {
```

Comment:
These changes aren't a functional change, I just happened to be looking at this function and find it easier to read with the inner enum matched on directly.